### PR TITLE
O2O test setup modified to work across boost version changes

### PR DESCRIPTION
**Pull request by ggovi**
#### PR description:

The workflows of integration tests for most of the O2Os are currently executed on special sqlite exports of the concerned production Tags, where only the second-last iov is copied.  This choice, implemented with the command "conddb copy --o2oTest", allows to limit the execution time of the O2O workflow to a reasonable time. However, the present implementation fails when the second-last iov points to a payload produced with a boost release incompatible (=more recent) with the underlying release boost version. Typically, this happening in CMSSW branches that are left behind with respect to the last release used in production for the O2Os.
This problem is the cause of issue https://github.com/cms-sw/cmssw/issues/33989

To overcome the problem, the o2oTest option of the function conddb copy has been modified, to select the highest iov in the source tag, pointing to a payload with boost version compatible with the underlying CMSSW release.

The new implementation fixes the above issue.

### CODE REVIEW

**Comment by ggovi** [comment]
please test

**Comment by ggovi** [comment]
please test

File: CondCore/Utilities/scripts/conddb [hunk 1]
```diff
@@ -739,6 +739,24 @@ def _get_last_frozen_since( session, tagName, fcsr=None ):
...
+    for r in q:
+        r= _rawdict(r)
+        if boost_version < r['boost_version']:
+            tlumi = r['run_number']<<32|0x1
+            time = (r['run_number'],tlumi,r['run_start_time'])
+            break
+    if timeType=='Run':
+        return time[0]
+    elif timeType=='Run':
+        return time[1]
```
Comment (francescobrivio) [inline CondCore/Utilities/scripts/conddb:755]: Are these supposed to be both `timeType == 'Run'` ?

**Comment by tvami** [comment]
test parameters:
  - addpkg = CondTools

File: CondCore/Utilities/scripts/conddb (hunk 1 above)
Comment (ggovi) [reply CondCore/Utilities/scripts/conddb]: yes. not relevant for the above issue (all O2Os populates either run or timestamp tags), but definitely logically wrong.

**Comment by ggovi** [comment]
please test

**Comment by ggovi** [comment]
test parameters:

addpkg = CondTools

**Comment by tvami** [comment]
+db

**Comment by perrotta** [comment]
+1

**Comment by ggovi** [comment]
@tvami a bit too fast :-) 
@perrotta 
 there is still one test failing in the 10_6_X build ( for some reason did not come up in my local testing ). I have ready the fix, but how do we deal with it? In principle I could add it directly to the 10_6_X PR (https://github.com/cms-sw/cmssw/pull/35442) but I would need to make a new PR (disjoint from 35443) for the master? Please let me know how to proceed...

**Comment by perrotta** [comment]
> @tvami a bit too fast :-) @perrotta there is still one test failing in the 10_6_X build ( for some reason did not come up in my local testing ). I have ready the fix, but how do we deal with it? In principle I could add it directly to the 10_6_X PR (#35442) but I would need to make a new PR (disjoint from 35443) for the master? Please let me know how to proceed...

The backport #35442 has far many more changes than the few lines that were coded here. Does the fix relate to those extra lines, or the ones in common?
In any case, if any fix is needed also for the master, please implement if in an additional PR.

**Comment by ggovi** [comment]
@perrotta Ah, right: the extra fixes in 10_6_X are coming from the code-checks + code-format, that was already executed here in the master! The new fix will be totally unrelated.  So - do I understand correctly:
- Add an extra commit in #35442, to provide a fix for the remaining failing test
- Port that fix in a new PR for master
Please confirm!

**Comment by perrotta** [comment]
> @perrotta Ah, right: the extra changes already appearing in 10_6_X are coming from the code-checks + code-format, that was already executed here in the master! The new fix will be totally unrelated. So - do I understand correctly:
> 
> * Add an extra commit in [O2O test setup modified to work across boost version changes [10_6_X] #35442](https://github.com/cms-sw/cmssw/pull/35442), to provide a fix for the remaining failing test
> * Port that fix in a new PR for master
> 
> Please confirm!

Correct @ggovi , please proceed

**Pull request merged**